pennylane/ops/op_math/pow_class.py

```diff
@@ -0,0 +1,285 @@
+    def data(self, new_data):
+        """Allows us to set base operation parameters."""
+        self.base.data = new_data
+
+    @property
+    def parameters(self):
+        return self.base.parameters
+
+    @property
+    def num_params(self):
+        return self.base.num_params
+
+    @property
+    def wires(self):
+        return self.base.wires
+
+    # pylint: disable=protected-access
+    @property
+    def _wires(self):
+        return self.base._wires
+
+    # pylint: disable=protected-access
+    @_wires.setter
+    def _wires(self, new_wires):
+        # used in a couple places that want to update the wires of an operator
+        # we should create a better way to set new wires in the future
+        self.base._wires = new_wires
+
+    @property
+    def num_wires(self):
+        return len(self.wires)
+
+    def queue(self, context=QueuingContext):
+        context.safe_update_info(self.base, owner=self)
+        context.append(self, owns=self.base)
+
+        return self
+
+    def label(self, decimals=None, base_label=None, cache=None):
+        z_string = format(self.z).translate(_superscript)
+        return self.base.label(decimals, base_label, cache=cache) + z_string
+
+    # pylint: disable=arguments-renamed, invalid-overridden-method
+    @property
+    def has_matrix(self):
+        return self.base.has_matrix
+
+    # pylint: disable=arguments-differ
+    @staticmethod
+    def compute_matrix(*params, base=None, z=0):
+        base_matrix = base.compute_matrix(*params, **base.hyperparameters)
+
+        if isinstance(z, int):
+            return qmlmath.linalg.matrix_power(base_matrix, z)
+
+        return fractional_matrix_power(base_matrix, z)
+
+    # pylint: disable=arguments-differ
+    @staticmethod
+    def compute_sparse_matrix(*params, base=None, z=0):
+        if isinstance(z, int):
+            base_matrix = base.compute_sparse_matrix(*params, **base.hyperparameters)
+            return base_matrix**z
+        raise SparseMatrixUndefinedError
+
+    def decomposition(self):
+        try:
+            return self.base.pow(self.z)
+        except PowUndefinedError as e:
+            if isinstance(self.z, int) and self.z > 0:
+                if QueuingContext.recording():
+                    return [apply(self.base) for _ in range(self.z)]
+                return [self.base.__copy__() for _ in range(self.z)]
+            # what if z is an int and less than 0?
+            # do we want Pow(base, -1) to be a "more fundamental" op
+            raise DecompositionUndefinedError from e
+
+    def diagonalizing_gates(self):
+        if isinstance(self.z, int):
+            return self.base.diagonalizing_gates()
+        # does this hold for non-integer z?
+        return super().diagonalizing_gates()
+
+    def eigvals(self):
+        base_eigvals = self.base.eigvals()
+        return [value**self.z for value in base_eigvals]
+
+    def generator(self):
+        return self.z * self.base.generator()
```

Comment:
Is there maybe some material that we could link to explaining (and potentially proving) that these two hold as such?

Reply:
$$
\left(e^{i \phi G}\right)^n = e^{i \phi (nG)}
$$
I guess I could add these to a docstring.